plugins/output-elasticsearch/src/main/scala/com/stratio/sparkta/plugin/output/elasticsearch/ElasticSearchOutput.scala

```diff
@@ -73,18 +94,54 @@ class ElasticSearchOutput(keyName: String,
 
   override val idField = properties.getString("idField", None)
 
-  override val defaultIndexMapping = properties.getString("indexMapping",
-    Some(DEFAULT_INDEX_TYPE))
+  override val mappingType = properties.getString("indexMapping", Some(DEFAULT_INDEX_TYPE))
+
+  override def setup: Unit = { createIndices }
+
+  private def createIndices = {
+    bcSchema.get.value.filter(tschema => (tschema.outputName == keyName)).foreach(tschemaFiltered => {
+      val tableSchemaTime = getTableSchemaFixedId(tschemaFiltered)
+      createIndexAcordingToSchema(tableSchemaTime.tableName, tableSchemaTime.schema)
+    })
+    elasticClient.close()
+  }
+
+  private def createIndexAcordingToSchema(tableName: String, schema: StructType) = {
+    elasticClient.execute {
+      create index tableName shards 1 replicas 0 mappings(
+        mappingType.get.toLowerCase as(getElasticsearchFields(schema))
+        )
+    }
+  }
 
-  override val indexMapping = getIndexType(defaultIndexMapping)
 
   override def doPersist(stream: DStream[(DimensionValuesTime, Map[String, Option[Any]])]): Unit = {
-    if (indexMapping.isDefined) persistDataFrame(stream)
+    persistDataFrame(stream)
   }
 
   override def upsert(dataFrame: DataFrame, tableName: String): Unit = {
-    val indexNameType = (tableName + "/" + indexMapping.get).toLowerCase
-    dataFrame.saveToEs(indexNameType, getSparkConfig(timeName, idField.isDefined || isAutoCalculateId))
+    val sparkConfig = getSparkConfig(timeName, idField.isDefined || isAutoCalculateId)
+    val indexNameType = (tableName + "/" + mappingType.get).toLowerCase
+    dataFrame.saveToEs(indexNameType, sparkConfig)
+  }
 
+
+
+  //scalastyle:off
+  def getElasticsearchFields(schema: StructType): Seq[TypedFieldDefinition] = {
+    schema.map(structField => structField.dataType match {
+      case org.apache.spark.sql.types.LongType => structField.name typed FieldType.LongType
+      case org.apache.spark.sql.types.DoubleType => structField.name typed FieldType.DoubleType
+      case org.apache.spark.sql.types.DecimalType() => structField.name typed FieldType.DoubleType
+      case org.apache.spark.sql.types.IntegerType => structField.name typed FieldType.IntegerType
+      case org.apache.spark.sql.types.BooleanType => structField.name typed FieldType.BooleanType
+      case org.apache.spark.sql.types.DateType => structField.name typed FieldType.DateType
+      case org.apache.spark.sql.types.TimestampType => structField.name typed FieldType.DateType
+      case org.apache.spark.sql.types.ArrayType(_,_) => structField.name typed FieldType.MultiFieldType
+      case org.apache.spark.sql.types.StringType => structField.name typed FieldType.StringType index "not_analyzed"
+      case _ => structField.name typed FieldType.BinaryType
+    })
```

Comment:
These are all the posibilities. These are the posible types due to PolicyFactory.rowTypeFromOption